core/src/services/moka/backend.rs

```diff
@@ -76,15 +93,52 @@ impl MokaBuilder {
 
     /// Sets the time to idle of the cache.
     ///
-    /// Refer to [`moka::sync::CacheBuilder::time_to_idle`](https://docs.rs/moka/latest/moka/sync/struct.CacheBuilder.html#method.time_to_idle)
+    /// Refer to [`moka::future::CacheBuilder::time_to_idle`](https://docs.rs/moka/latest/moka/sync/struct.CacheBuilder.html#method.time_to_idle)
     pub fn time_to_idle(mut self, v: Duration) -> Self {
         if !v.is_zero() {
             self.config.time_to_idle = Some(v);
         }
         self
     }
 
-    /// Set root path of this backend
+    /// Configure the cache builder with a closure.
+    ///
+    /// Refer to [`moka::future::CacheBuilder`](https://docs.rs/moka/latest/moka/future/struct.CacheBuilder.html)
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use std::sync::Arc;
+    /// # use std::time::Duration;
+    /// # use moka::notification::RemovalCause;
+    /// # use opendal::services::MokaConfig;
+    /// # use opendal::services::MokaValue;
+    /// # use opendal::Configurator;
+    /// # use log::debug;
+    /// let builder = MokaConfig::default().into_builder();
+    /// let moka = builder.configure(|builder| {
+    ///     builder
+    ///         .name("demo")
+    ///         .max_capacity(1000)
+    ///         .time_to_live(Duration::from_secs(300))
+    ///         .weigher(|k, v| (k.len() + v.content.len()) as u32)
+    ///         .eviction_listener(|k: Arc<String>, v: MokaValue, cause: RemovalCause| {
+    ///             debug!(
+    ///                 "moka cache eviction listener, key = {}, value = {:?}, cause = {:?}",
+    ///                 k.as_str(), v.content.to_vec(), cause
+    ///             );
+    ///         })
+    /// });
+    /// ```
+    pub fn configure<F>(mut self, f: F) -> Self
```

Comment:
I'm thinking of expose a `from_builder` API which accepts `moka::future::CacheBuilder<String, MokaValue>` directly, what do you think? In this way, the only thing we need to expose is `MokaValue`. Expose the type alias `MokaCacheBuilder` also looks fine to me.

Reply:
You mean like this?

```rust
pub fn from_builder(mut self, builder: MokaCacheBuilder<String, MokaValue>) -> Self { 
    self.builder = builder;
    self
}
```

Honestly, I think the name might be a bit confusing. If I assemble the operator with the following api:

```rust
let builder = MokaConfig::default().into_builder().from_builder(MokaCacheBuilder::default());
let operator = Operator::new(builder)?.finish();

// Of course you can assemble the operator like  this
let builder = Moka::default().from_builder(MokaCacheBuilder::default());
let operator = Operator::new(builder)?.finish();
```

Reply:
I agree with you that `from_builder` is not a good name. How about providing a `new(MokaCacheBuilder)`?